nilearn/glm/second_level/second_level.py

```diff
@@ -650,77 +662,133 @@ def non_parametric_inference(second_level_input, confounds=None,
+        If True, a constant column is added to the confounding variates
+        unless the tested variate is already the intercept.
+        Default=True.
+
+    n_perm : :obj:`int`, optional
+        Number of permutations to perform.
+        Permutations are costly but the more are performed, the more precision
+        one gets in the p-values estimation. Default=10000.
+
+    two_sided_test : :obj:`bool`, optional
+        If True, performs an unsigned t-test. Both positive and negative
+        effects are considered; the null hypothesis is that the effect is zero.
+        If False, only positive effects are considered as relevant. The null
+        hypothesis is that the effect is zero or negative.
+        Default=False.
+
+    random_state : :obj:`int` or None, optional
+        Seed for random number generator, to have the same permutations
+        in each computing units.
+
+    n_jobs : :obj:`int`, optional
+        Number of parallel workers.
+        If -1 is provided, all CPUs are used.
+        A negative number indicates that all the CPUs except (abs(n_jobs) - 1)
+        ones will be used. Default=1.
+
+    verbose : :obj:`int`, optional
         Verbosity level (0 means no message). Default=0.
 
+    threshold : None or :obj:`float`, optional
+        Cluster-forming threshold in p-scale.
+        This is only used for cluster-level inference.
+        If None, no cluster-level inference will be performed.
+        Default=None.
+
+        .. warning::
+
+            Performing cluster-level inference will increase the computation
+            time of the permutation procedure.
+
+        .. versionadded:: 0.9.2
+
     Returns
     -------
-    neg_log_corrected_pvals_img : Nifti1Image
+    neg_log10_vfwe_pvals_img : Nifti1Image
         The image which contains negative logarithm of the
-        corrected p-values.
+        voxel-level FWER-corrected p-values.
+
+        .. note::
+            This is returned if ``threshold`` is None (the default).
+
+    outputs : :obj:`dict`
+        Output images, organized in a dictionary.
+        Each image is 3D/4D, with the potential fourth dimension corresponding
+        to the regressors.
+
+        .. note::
+            This is returned if ``threshold`` is not None.
+
+        .. versionadded:: 0.9.2
+
+        Here are the keys:
+
+        =============== =======================================================
+        key             description
+        =============== =======================================================
+        t               T-statistics associated with the significance test of
+                        the n_regressors explanatory variates against the
+                        n_descriptors target variates.
+        logp_max_t      Negative log10 family-wise error rate-corrected
+                        p-values corrected based on the distribution of maximum
+                        t-statistics from permutations.
```

Comment:
These are also only returned if `threshold` is not `None` correctly ? You explicitly include that in the description of the other two keys, so I just wanted to check !

Reply:
This is the standard voxel-level-corrected map, so it will be returned as long as `n_iters > 0`.

Reply:
But this object is noted to only be returned when `threshold` is not `None` ? I think that's the source of my confusion !

Reply:
Ah, I see. Yes, in this PR the dictionary output is only returned if threshold is defined. In the TFCE PR (#3196), I add an `output_type` parameter that lets users directly select the new structure.